base/linalg/arnoldi.jl

```diff
@@ -237,7 +237,7 @@ function _eigs(A, B;
     end
 
     # Refer to ex-*.doc files in ARPACK/DOCUMENTS for calling sequence
-    matvecA(x) = A * x
+    matvecA!(y, x) = A_mul_B!(y, A, x)
     if !isgeneral           # Standard problem
         matvecB = x -> x
```

Comment:
I tried that but <a href="https://github.com/JuliaLang/julia/pull/16294/files#diff-503c2ea34331f7c62b228dd425a7539eR90">here</a> we have `y[:] = solveSI(matvecB(x))`, and that seemed to slow things down in some cases.

I would have guessed that `copy!(y,x)` would be effectively result in the identity when `y === x` (maybe it is if `memmove` takes care of that, but I saw a performance hit anyway).